ui/app/components/auth/form-template.hbs

```diff
@@ -40,22 +40,16 @@
         <:authSelectOptions>
           <div class="has-bottom-margin-m">
             {{#if this.showCustomAuthOptions}}
-              {{! URL contains "with" query param and it references a visible mount }}
-              {{! treat it as a "preferred" mount and hide all other tabs }}
-              {{#if @directLinkData.hasMountData}}
-                {{#let @directLinkData as |mount|}}
-                  <Auth::SingleMount
-                    @description={{mount.description}}
-                    @path={{mount.path}}
-                    @shouldRenderPath={{not-eq @selectedAuthMethod "token"}}
-                    @type={{this.displayName mount.type}}
-                  />
-                {{/let}}
+              {{#if @directLinkData.isVisibleMount}}
+                {{! URL contains a "with" query param that references a mount with listing_visibility="unauth" }}
+                {{! Treat it as a "preferred" mount and hide all other tabs }}
+                <Auth::MountsDisplay
+                  @mounts={{array @directLinkData}}
```

Comment:
went back and forth on whether I should also pass in a `@mount` (singular) arg but keeping the type consistent and always passing an array of mount objects seemed the most straightforward.